docs/source/design/testing.rst

```diff
@@ -0,0 +1,99 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_testing:
+
+**********************************
+Continuous integration and testing
+**********************************
+
+.. topic:: Specification scope and status
+
+   This specification describes the current implementation.
+
+DataLad is tested using a pytest-based testsuite that is run locally and via continuous integrations setups.
+Code development should ensure that old and new functionality is appropriately tested.
+The project aims for good unittest coverage (at least 80%).
+
+Running tests
+=============
+
+``datalad/tests`` contains tests for the core portion of the project.
+More tests are provided under corresponding submodules in ``tests/`` subdirectories to simplify re-running the tests concerning that portion of the codebase.
+The test suite is run using
+
+.. code-block:: bash
+
+   pip install -e .[tests]
+   python -m pytest -c tox.ini datalad
+   # or, with coverage reports
+   python -m pytest  -c tox.ini --cov=datalad datalad
+
+Individual tests can be run using a path to the test file, followed by two colons and the test name:
+
+.. code-block:: bash
+
+    python -m pytest datalad/core/local/tests/test_save.py::test_save_message_file
+
+The set of to-be-run tests can be further sub-selected with environment variable based configurations that enable tests based on their :ref:`decorators`, or pytest-specific parameters.
+Invoking a test run using ``DATALAD_TESTS_KNOWNFAILURES_PROBE=True pytest datalad``, for example, will run tests marked as known failures whether or not they still fail.
+See section :ref:`configuration` for all available configurations.
+Invoking a test run using ``DATALAD_TESTS_SSH=1 pytest -m xfail -c tox.ini datalad`` will run only those tests marked as `xfail <https://docs.pytest.org/en/latest/how-to/skipping.html>`_.
+
+Local setup
+-----------
+Local test execution usually requires a local installation with all development requirements. It is recommended to either use a `virtualenv <https://virtualenv.pypa.io/en/latest/>`_, or `tox <https://tox.wiki/en/latest/>`_ via a ``tox.ini`` file in the code base.
+
+CI setup
+--------
+At the moment, Travis-CI, Appveyor, and GitHub Workflows exercise the tests battery for every PR and on the default branch, covering different operating systems, Python versions, and file systems.
+Tests should be ran on the oldest, latest, and current stable Python release.
+The projects uses https://codecov.io for an overview of code coverage.
+
+
+Writing tests
+=============
+
+Additional functionality is tested by extending existing similar tests with new test cases, or adding new tests to the respective test script of the module.
+Test helper functions assisting various general and DataLad specific assertions as well the construction of test directories and files can be found in ``datalad/tests/utils_pytest.py`` and ``datalad/tests/utils.py``.
```

Comment:
Don't mention the nose-based utils. They are supposed to be removed and we should not make people trying to reintroduce usage or enhancing them.
```suggestion
Test helper functions assisting various general and DataLad specific assertions as well the construction of test directories and files can be found in ``datalad/tests/utils_pytest.py``.
```